### PR TITLE
[DOC] Update Python UDFs notebook

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Updates a rather outdated IPython notebook describing the UDF frameworks in cuDF python.

- Fixes up outdated API calls that don't work with this branch.
- Removes the old section about null handling, which leverages functions we have removed at this point.
- Adds a new section about null handling through our new framework.

### CODE REVIEW

**Comment by quasiben** [comment]
Thanks for doing this @brandon-b-miller -- looks great

**Review by quasiben — APPROVED** [review]

File: docs/cudf/source/guide-to-udfs.ipynb [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "- CuPy NDArrays\n",
```
Comment (shwina) [inline docs/cudf/source/guide-to-udfs.ipynb:22]: <p>We publish docs separately for different cuDF versions, so it's not strictly necessary to include versioning information in the docs. Ditto below.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/8810/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (shwina) [inline docs/cudf/source/guide-to-udfs.ipynb:22]: <p>As above, I think this section can just be <code>N/A support</code>  without references to the behavior in previous versions. With time, previous behaviour will become less relevant with time and likely we won't get a chance to update the docs.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/8810/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

**Review by shwina — APPROVED** [review]

**Comment by brandon-b-miller** [comment]
@gpucibot merge

**Pull request merged**